lib/SILOptimizer/Mandatory/MandatoryInlining.cpp

```diff
@@ -481,6 +481,12 @@ class ClosureCleanup {
       if (!operandVal->getType().is<SILFunctionType>())
         continue;
 
+      // Skip debug instructions
```

Comment:
This check should be done at the function entry.

Can you add a bit more information in the comment? E.g. Add the info what you wrote in the PR description.
Also, you should mention that the special handling of debug instruction is done in eraseFromParentWithDebugInsts.